### ISSUE TITLE
v1.0.0-rc.9: How to use --include-groups

### ISSUE THREAD

**Issue by halfinhalfout** [awslabs/ssosync#41]
**Describe the bug**
This might be a misunderstanding of how to use `--include-groups` on my part, a code bug, or a problem w/ the objects in my GSuite External IdP. I'm not sure.

I've got something like 5 groups, containing around 100 users, that I want to ssosync. Importing all of my GSuite users/groups isn't an option as I have thousands of those, most of whom don't need access to AWS.

What is the syntax to use `--include-groups`? I've not been able to get it to work.

When I try 1 or more comma separated values like `officetech` or `officetech,webops` I receive a **Error 400: Invalid Input: query, invalid** response. Eg.,
```
./ssosync --debug --access-token "REDACTED" --endpoint "REDACTED" --google-admin "redacted@redacted" --google-credentials "~/Documents/ssosync.google.credentials.json" --include-groups "officetech"
INFO[0000] Syncing AWS users and groups from Google Workspace SAML Application
INFO[0000] syncing                                       sync_method=groups
DEBU[0000] get google groups                             query=
FATA[0000] googleapi: Error 400: Invalid Input: query, invalid
``` 

However, I am able to import my **officetech** Group and it's users with `--group-match "name:officetec*"`, and the Group and it's users are properly imported to AWS SSO.

I don't think `--group-match` meets my use case, however, since only the last incidence of that switch appears to be respected. For example, If I do `--group-match "name:webop*"  --group-match "name:officetec*"` , ssosync appears to only attempt to import the **officetech* Group/users.

**Expected behavior**
The following would import exactly 3 Groups, and their users (officetech, webops & thirdgroup):
```
./ssosync --debug --access-token "REDACTED" --endpoint "REDACTED" --google-admin "redacted@redacted" --google-credentials "~/Documents/ssosync.google.credentials.json" --include-groups "officetech,webops,thirdgroup"
```

**Comment by halfinhalfout** [comment]
Oh, is this because I _need_ some sort of `--group-match`, even when using `--include-groups`, due to https://github.com/awslabs/ssosync/issues/40 ?

**Comment by christiangda** [comment]
@halfinhalfout there are two `sync methods`

```
-s, --sync-method string          Select the sync method to use (users_groups|groups) (default "groups")
```

by default the behaviour is 'groups', which mean that you only want to sync groups and their members, additionally, you can use the filter for groups.

when you use `--sync-method "groups" ` the parameter `--group-match` doesn't work, so, if you want to sync all users and only groups into `--group-match "g1,g2,g..."` you need to set 

```
--sync-method "users_groups" 
```

**Comment by christiangda** [comment]
@halfinhalfout I never tested this pattern (two differents groups prefix)

```
--group-match "name:webop*" --group-match "name:officetec*"
```

So, it is sure that this doesn't work, I can try to add this functionality.

**Comment by joshuachong** [comment]
@halfinhalfout
the confusion starts with the sync_method which has two options `groups` (default) and `user_groups`. The include-groups is applicable when performing user_group matching (which includes users, and then groups).  You can specify multiple include-groups which you can limit to the 5 that you intend to.

@halfinhalfout @christiangda 
i'm interested to find out if you have a certain structure that you name your Groups? If a regex pattern list and plus a list of ignore groups will help out?

l've heard others that are hitting API rate limits due to large group memberships so have to be plan around that area too.

**Comment by evilr00t** [comment]
@christiangda looks like `--ignore-groups` does not work with`-s groups`

```
evilroot in ~ λ ssosync --log-level debug --ignore-groups all@xyz.com --google-admin xyz@xyz.com -s groups -c creds.json
INFO[0000] Syncing AWS users and groups from Google Workspace SAML Application
INFO[0000] syncing                                       sync_method=groups
DEBU[0000] get google groups                             query=
FATA[0001] googleapi: Error 400: Invalid Input: query, invalid
evilroot in ~ λ ssosync --log-level debug --ignore-groups all@xyz.com --google-admin xyz@xyz.com -s users_groups -c creds.json
INFO[0000] Syncing AWS users and groups from Google Workspace SAML Application
INFO[0000] syncing                                       sync_method=users_groups
DEBU[0000] get deleted users
INFO[0000] deleting google user                          email=xyz@xyz.com
DEBU[0001] get active google users
DEBU[0002] finding user                                  email=xyz@xyz.com
DEBU[0005] get google groups                             query=
FATA[0005] googleapi: Error 400: Invalid Input: query, invalid
```

**Comment by joshuachong** [comment]
Hi @evilr00t 
The `Error 400` is due to the filter query being empty. You will need to include the `-g` and `-m` to ensure the filter is not empty.

**Comment by christiangda** [comment]
@halfinhalfout this was fixed in PR #44

**Comment by halfinhalfout** [comment]
Thank you @christiangda & @joshuachong 

This was a a proof of concept investigation at my org that I was driving. I lost access to an environment to test this in shortly after I submitted this issue. I expect to have access again sometime in the first quarter of 2022. Based on your comments here and the new README info in PR [#44](https://github.com/awslabs/ssosync/pull/44), I have an idea of how I might be able to do what I need.

Do you think this is a viable way of syncing only specific Groups (and all of their members), when my groups won't easily be matchable by a sane regex expression, and I have a majority of Groups/Users that I do _not_ need to sync?

1. Add a dummy user to Gsuite (eg., ssosync-dummy@domain.com)
2. Join that dummy user to each Group we want sync'd
3. Run ssosync with options something like: 
    ````
      ssosync --debug --access-token "REDACTED" --endpoint "REDACTED" --google-admin "REDACTED@domain.com" --google-credentials "ssosync.google.credentials.json" --group-match "memberKey=ssosync-dummy@domain.com" --sync-method "groups"
     ````

That's based upon the assumption that `--group-match` supports the `memberKey=` syntax of https://developers.google.com/admin-sdk/directory/v1/guides/search-groups

Does look like it would work to you?

Thanks!

**Comment by scristian** [comment]
> @halfinhalfout this was fixed in PR #44

Yes, in -rc10 is working fine now.
